### ISSUE TITLE
read.met() has temperature 10X the proper value

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#799]
Funny that I didn't notice this in 2011 in the SLE work ... guess we only used pressure and speeds.

**Comment by dankelley** [comment]
Also the pressure is getting multiplied by 10, which is wrong ... well, it gives millibars so that makes some sense I guess but since I'm fixing temperature I may as well fix this, so the unit will be in kPa as is probably more modern.

**Comment by dankelley** [comment]
Fixed in `develop` commit 95da60af287e126b8e7b5dff7326bc7c3f0bfad6

**Comment by richardsc** [comment]
Considering that when I try and run the current oce on the files that I originally downloaded for SLEIWEX, I get as below, I expect that the format has changed:
```r
> library(oce)
Loading required package: gsw
> d <- read.oce('riviere-du-loups-eng-hourly-07012008-07312008.csv')
Error in if (grep("*.nc$", filename)) return("netcdf") : 
  argument is of length zero
> d <- read.oce('ile-rouge-eng-hourly-07012008-07312008.csv')
Error in if (grep("*.nc$", filename)) return("netcdf") : 
  argument is of length zero
```

The old files have a `head` that looks like:
```
"Station Name","RIVIERE-DU-LOUP"
"Province","QUEBEC"
"Latitude","47.81"
"Longitude","-69.55"
"Elevation","146.50"
"Climate Identifier","7056616"
"WMO Identifier","71715"
"TC Identifier","WNH"
"All times are specified in Local Standard Time (LST). Add 1 hour to adjust for Daylight Saving Time where and when it is observed."

"Legend"
"M","Missing"
"E","Estimated"
"NA","Not Available"
"**","Partner data that is not subject to review by the National Climate Archives"

"Date/Time","Year","Month","Day","Time","Data Quality","Temp (∞C)","Temp Flag","Dew Point Temp (∞C)","Dew Point Temp Flag","Rel Hum (%)","Rel Hum Flag","Wind Dir (10's deg)","Wind Dir Flag","Wind Spd (km/h)","Wind Spd Flag","Visibility (km)","Visibility Flag","Stn Press (kPa)","Stn Press Flag","Hmdx","Hmdx Flag","Wind Chill","Wind Chill Flag","Weather"
"2008-07-01 00:00","2008","07","01","00:00"," ","9.40","","9.10","","98.00","","11.00","","4.00","","","","98.75","","","","","","NA"
"2008-07-01 01:00","2008","07","01","01:00"," ","10.50","","10.20","","98.00","","35.00","","2.00","","","","98.77","","","","","","NA"
"2008-07-01 02:00","2008","07","01","02:00"," ","9.90","","9.60","","98.00","","18.00","","2.00","","","","98.79","","","","","","NA"
```

**Comment by richardsc** [comment]
N.B. I'm not suggesting that we make the current oce read those old files -- that was just an example of the fact that the EnvCanada format has changed since then.

**Comment by dankelley** [comment]
That `grep()` error was a bug that entered a few days ago. Oddly, I cannot find in my code the spot where  `oceMagic()` recognizes the old met file type.  I could try to make it check for first tokens `"Station Name"` and filename ending in `.csv` I guess, although I worry that might conflict with some CTD type data.  Maybe I need to see if `"WMO Identifier"` is in the header. 

I just checked, and `read.met()` (in `develop` commit ba218a2e6c9854631bec690d42b41fc575aef13f) handles both the old file you mention and also Shannon's new file. I'm not sure the file format has changed much, therefore.

**Question** any thoughts on my idea for `oceMagic()`? The ideas (please vote) are
* [ ] check for filename ending `.csv` and for contents line starting `Station Name`
* [ ] check for filename ending `.csv` and a header line with `WMO identifier`
* [ ] check for filename ending `.csv` and for contents line starting `Station Name` and a header line with `WMO identifier` (DK vote)

PS. reopening for a while until I settle on whether to make `oceMagic()` as suggested.

**Comment by richardsc** [comment]
Voted. I wouldn't bother making any changes to `oceMagic()` for a file that was downloaded 6 years ago, if it works on the files currently pulled from the website.

**Comment by dankelley** [comment]
Sorry, I didn't see your vote.  I just pushed to `develop` a version in which `oceMagic()` recognizes the `csv` ending and the content `WMO Identifier`. It turns out that no other tests were done on `csv` ending, so this won't cause any conflicts.  Note that `oceMagic()` does not conflict with `read.*()` functions, so it's possible for any `read.*()` function to handle `.csv` files.

PS. I just did a build-check in Rstudio and there's a bug in the `plotProfile()` code-doc match ... I'll push a solution in a minute or two.
